torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -8821,14 +8821,63 @@ def celu_hook(module, _input, output):
             self.assertEqual(ddp_grads[0], local_model.fc.weight.grad)
             self.assertEqual(ddp_grads[1], local_model.fc.bias.grad)
 
-        def _test_hook_pickling(self, hook, hook_state):
+        def _test_hook_pickling_logs(
+            self,
+            hook,
+            hook_state,
+            restore_msg,
+        ):
+            chkpt_file = tempfile.gettempdir() + "/checkpoint.pt"
```

Comment:
@rohan-varma is there any reason why we do not use a context manager to manage the temp file? Are temp files/directories automatically cleaned up by the OS even if we do not call `os.remove()`?

(@aovladi not related to this PR)